AdobeMediaGallery/Model/Asset/Command/GetListByPath.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeMediaGallery\Model\Asset\Command;
+
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterface;
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterfaceFactory;
+use Magento\AdobeMediaGalleryApi\Model\Asset\Command\GetListByPathInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Exception\IntegrationException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class GetListByIds
+ */
+class GetListByPath implements GetListByPathInterface
+{
+    private const TABLE_MEDIA_GALLERY_ASSET = 'media_gallery_asset';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var AssetInterface
+     */
+    private $mediaAssetFactory;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * GetListByPath constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param AssetInterfaceFactory $mediaAssetFactory
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        AssetInterfaceFactory $mediaAssetFactory,
+        LoggerInterface $logger
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->mediaAssetFactory = $mediaAssetFactory;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Return media asset asset list
+     *
+     * @param string $mediaFilePath
+     *
+     * @return array
+     * @throws IntegrationException
+     */
+    public function execute(string $mediaFilePath): array
```

Comment:
Considering that path is unique, I'd rename the command to `GetByPath` and return `AssetInterface`